otelcollector/configmapparser/tomlparser-default-scrape-settings.rb

```diff
@@ -93,27 +98,48 @@ def populateSettingValuesFromConfigMap(parsedConfig)
       @kappiebasicEnabled = parsedConfig[:kappiebasic]
       puts "config::Using configmap scrape settings for kappiebasic: #{@kappiebasicEnabled}"
     end
+    if !parsedConfig[:"kube-controller-manager"].nil?
+      @kubecontrollermanagerEnabled = parsedConfig[:"kube-controller-manager"]
+      puts "config::Using configmap scrape settings for kube-controller-manager: #{@kubecontrollermanagerEnabled}"
+    end
+    if !parsedConfig[:"kube-scheduler"].nil?
+      @kubeschedulerEnabled = parsedConfig[:"kube-scheduler"]
+      puts "config::Using configmap scrape settings for kube-scheduler: #{@kubeschedulerEnabled}"
+    end
+    if !parsedConfig[:"kube-apiserver"].nil?
+      @kubeapiserverEnabled = parsedConfig[:"kube-apiserver"]
+      puts "config::Using configmap scrape settings for kube-apiserver: #{@kubeapiserverEnabled}"
+    end
+    if !parsedConfig[:"cluster-autoscaler"].nil?
+      @clusterautoscalerEnabled = parsedConfig[:"cluster-autoscaler"]
+      puts "config::Using configmap scrape settings for cluster-autoscaler: #{@clusterautoscalerEnabled}"
+    end
+    if !parsedConfig[:"etcd"].nil?
+      @etcdEnabled = parsedConfig[:"etcd"]
+      puts "config::Using configmap scrape settings for etcd: #{@etcdEnabled}"
+    end
 
     windowsDaemonset = false
     if ENV["WINMODE"].nil? && ENV["WINMODE"].strip.downcase == "advanced"
       windowsDaemonset = true
     end
 
+    ccpmetricsEnabled = !kubecontrollermanagerEnabled && !kubeschedulerEnabled && !kubeapiserverEnabled && !clusterautoscalerEnabled && !etcdEnabled
```

Comment:
maybe? taking a look at the code below where this is used seems to be used for the advanced mode and current logic evaluates to all negative. Let me know what you think
<img width="1347" alt="image" src="https://github.com/sgeannina/prometheus-collector/assets/79080068/3ea6518b-cb9d-4d2e-816e-6e1f0a31ee5f">

Reply:
yes but its also negated in the below code by **!**ccpmetricsEnabled, which reverses each of the composite checks to be just: kubecontrollermanagerEnabled (without a negating !)

We want ccpmetricsEnabled to evaluate to true when any of kubecontrollermanager, kubescheduler, kubeapiserver, clusterautoscaler or etcd are enabled, right? 

So we can either reverse the logic to:
kubecontrollermanagerEnabled || kubeschedulerEnabled... etc, 
 
or rename it to ccpmetrics**Disabled** = !kubecontrollermanagerEnabled && !kubeschedulerEnabled, and remove the ! from the && ccpmetricsDisabled check in the code below (I don't like this approach as much as keeping it named ccpmetricsEnabled, because boolean logic for things that are !disabled are mistake prone)

Reply:
good catch! I'll change it